SpinninMonke/Assets/Scripts/Player/Movement.cs

```diff
@@ -0,0 +1,82 @@
+using UnityEngine;
+
+namespace Player
+{
+    [RequireComponent(typeof(Rigidbody))]
+    public class Movement : MonoBehaviour
+    {
+        public float maxSpeed;
+
+        [Range(0.0f, 0.99f)]
+        public float velocitySmoothness;
+
+        public float rotationSpeed;
+
+        private const float MaxVelocitySmoothness = 1.0f;
+
+        private ThirdPersonCamera _camera;
+        private Rigidbody _rigidbody;
+
+        private Vector3 _movementDirection;
+        private bool ShouldMove => _movementDirection != Vector3.zero;
```

Comment:
float equality checks are not reliable, you should check if magnitude of the vector is above some threshold